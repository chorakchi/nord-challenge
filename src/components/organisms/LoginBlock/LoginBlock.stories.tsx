import { ComponentStory, ComponentMeta } from "@storybook/react";

import { LoginBlock } from "./LoginBlock";

export default {
  title: "organisms/LoginBlock",
  component: LoginBlock,
  parameters: {
    docs: {
      description: {
        component: "",
      },
    },
    backgrounds: {
      default: "light",
    },
  },
} as ComponentMeta<typeof LoginBlock>;

const Template: ComponentStory<typeof LoginBlock> = (args) => (
  <LoginBlock {...args} />
);

export const LoginBlock_ = Template.bind({});
LoginBlock_.args = {};
