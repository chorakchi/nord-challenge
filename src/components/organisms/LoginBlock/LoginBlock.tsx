import { TypographyProps } from "./LoginBlock.interface";
import { TextField, Form, Typography, Button } from "components";
export const LoginBlock: React.FC<TypographyProps> = ({
  message,
  onClickSignIn,
  onChangeFields,
  ...props
}: TypographyProps) => {
  return (
    <div {...props} className="max-w-md w-full space-y-8">
      <Typography>üè†</Typography>
      <Typography>Homework - Sign in</Typography>
      <Form
        onSubmit={(e) => {
          e.preventDefault();
          onClickSignIn({
            username: e.target[1].value,
            password: e.target[2].value,
          });
        }}
      >
        <TextField.Group>
          <TextField
            id="username"
            name="username"
            type="text"
            label="Username"
            required
            placeholder="Username"
            onChange={onChangeFields}
          />
          <TextField
            id="password"
            name="password"
            type="password"
            label="Password"
            autoComplete="current-password"
            required
            onChange={onChangeFields}
          />
        </TextField.Group>
        <Button type="submit">Sign in</Button>
        <Typography>{message}</Typography>
      </Form>
    </div>
  );
};
