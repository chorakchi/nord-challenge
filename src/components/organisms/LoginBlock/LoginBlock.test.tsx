import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { LoginBlock } from "./LoginBlock";

describe("LoginBlock", () => {
  it("should render the LoginBlock", () => {
    const subject = shallow(<LoginBlock onClickSignIn={()=>{}}/>);
    expect(subject).toHaveLength(1);
  });

  it("should render correctly and match to the Snapshot", () => {
    const subject = renderer.create(<LoginBlock onClickSignIn={()=>{}}/>).toJSON();
    expect(subject).toMatchSnapshot();
  });
});
