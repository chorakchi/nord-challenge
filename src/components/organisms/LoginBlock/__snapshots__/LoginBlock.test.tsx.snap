// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginBlock should render correctly and match to the Snapshot 1`] = `
<div
  className="max-w-md w-full space-y-8"
>
  <p
    className="mt-6 text-center text-3xl font-extrabold text-gray-900"
  >
    üè†
  </p>
  <p
    className="mt-6 text-center text-3xl font-extrabold text-gray-900"
  >
    Homework - Sign in
  </p>
  <form
    action="#"
    className="mt-8 space-y-6"
    method="POST"
    onSubmit={[Function]}
  >
    <input
      name="remember"
      type="hidden"
      value="true"
    />
    <div
      className="rounded-md shadow-sm -space-y-px"
    >
      <label
        className="sr-only"
        htmlFor="username"
      >
        Username
      </label>
      <input
        autoComplete="username"
        className="appearance-none rounded-none relative block w-full px-3 py-2 border border-gray-300 placeholder-gray-500 text-gray-900 rounded-t-md focus:outline-none focus:ring-indigo-500 focus:border-indigo-500 focus:z-10 sm:text-sm"
        id="username"
        name="username"
        placeholder="Username"
        required={true}
        type="text"
      />
      <label
        className="sr-only"
        htmlFor="password"
      >
        Password
      </label>
      <input
        autoComplete="current-password"
        className="appearance-none rounded-none relative block w-full px-3 py-2 border border-gray-300 placeholder-gray-500 text-gray-900 rounded-t-md focus:outline-none focus:ring-indigo-500 focus:border-indigo-500 focus:z-10 sm:text-sm"
        id="password"
        name="password"
        placeholder="Password"
        required={true}
        type="password"
      />
    </div>
    <button
      className="group relative w-full flex justify-center py-2 px-4 border border-transparent text-sm font-medium rounded-md text-white bg-indigo-600 hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
      type="submit"
    >
      Sign in
    </button>
    <p
      className="mt-6 text-center text-3xl font-extrabold text-gray-900"
    />
  </form>
</div>
`;
